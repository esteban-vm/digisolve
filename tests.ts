import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
