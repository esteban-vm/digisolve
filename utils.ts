export const formatPrice = (price: number) => {
  return new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
    maximumFractionDigits: 0,
  }).format(price)
}
