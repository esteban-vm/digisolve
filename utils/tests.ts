import type { ReactElement } from 'react'
import { create, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON()
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
