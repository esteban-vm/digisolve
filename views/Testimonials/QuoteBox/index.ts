export { default, type QuoteProps } from './QuoteBox'
