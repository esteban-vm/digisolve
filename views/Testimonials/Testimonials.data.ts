import type { PropsWithImage } from '@/types'
import type { TestimonialProps } from './Testimonial'

const text = 'Voluptates hic pariatur reiciendis minus vel labore dolorum.'
const width = 72
const height = 72

const testimonials: PropsWithImage<TestimonialProps>[] = [
  {
    id: 'c2f0fc36-aad3-4e98-9ff8-61a30257d41c',
    customer: 'Herman Gibson',
    text,
    image: { src: '/img/client-1.jpg', width, height },
  },
  {
    id: 'efc0be82-fe34-4df9-8821-cf195a75da1d',
    customer: 'Peter Kemmer',
    text,
    image: { src: '/img/client-2.jpg', width, height },
  },
  {
    id: 'c679bf1c-1387-4aa0-9eec-ab2d28d79310',
    customer: 'Naomi Stoltenberg',
    text,
    image: { src: '/img/client-3.jpg', width, height },
  },
]

export default testimonials
