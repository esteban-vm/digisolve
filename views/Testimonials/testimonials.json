[
  {
    "image": "/img/client-1.jpg",
    "text": "Aaron Cruz",
    "height": 72,
    "width": 72
  },
  {
    "image": "/img/client-2.jpg",
    "text": "Jeff Jones",
    "height": 72,
    "width": 72
  },
  {
    "image": "/img/client-3.jpg",
    "text": "Helen Smith",
    "height": 72,
    "width": 72
  }
]
