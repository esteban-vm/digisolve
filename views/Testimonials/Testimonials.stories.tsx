import type { Meta } from '@storybook/react'
import Testimonials from './Testimonials'

export default {
  component: Testimonials,
  title: 'Views/Testimonials',
} as Meta<typeof Testimonials>

export const TestimonialsView = () => <Testimonials />
