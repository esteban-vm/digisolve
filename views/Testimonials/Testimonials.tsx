import type { FunctionalComponent } from '@/types'
import { client1, client2, client3 } from '@/utils/images'
import { styled } from '@/utils/styles'
import QuoteBox, { type QuoteProps } from './QuoteBox'

export const quotes: QuoteProps[] = [
  { name: 'Aaron Cruz', photo: client1 },
  { name: 'Jeff Jones', photo: client2 },
  { name: 'Helen Smith', photo: client3 },
]

const TestimonialsComponent: FunctionalComponent = (props) => {
  return (
    <section {...props}>
      <div className='row'>
        <h2>Our Testimonials</h2>
      </div>
      <div className='row'>
        {quotes.map((quote, index) => (
          <QuoteBox key={index} {...quote} role='listitem' />
        ))}
      </div>
    </section>
  )
}

const Testimonials = styled(TestimonialsComponent)`
  background-image: linear-gradient(var(--color-overlay3), var(--color-overlay4)), url('/img/testimonial-bg.jpg');
  background-size: cover;
  background-attachment: fixed;
  text-align: center;
  color: var(--color-white);

  h2 {
    font-size: 250%;
    margin-bottom: 20px;
    margin-top: 150px;
  }
`

export default Testimonials
