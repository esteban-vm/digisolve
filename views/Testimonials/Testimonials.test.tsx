import { render, cleanup, screen, create } from '@/utils/tests'
import Testimonials, { quotes } from './Testimonials'

describe('ðŸ§ª <Testimonials /> test cases:', () => {
  describe('there should be:', () => {
    beforeEach(() => {
      render(<Testimonials />)
    })

    afterEach(cleanup)

    it('a heading', () => {
      const heading = screen.getByRole('heading', { name: /^our testimonials$/i })
      expect(heading).toBeInTheDocument()
      expect(heading).toBeVisible()
    })

    it(`${quotes.length} quote boxes`, () => {
      const numberOfQuotes = quotes.length
      const quoteElements = screen.getAllByRole('listitem')
      const quotePhotos = screen.getAllByRole('img')

      expect(quoteElements).toHaveLength(numberOfQuotes)
      expect(quotePhotos).toHaveLength(numberOfQuotes)

      for (let index = 0; index < numberOfQuotes; index++) {
        const quoteElement = quoteElements[index]
        expect(quoteElement).toBeInTheDocument()
        expect(quoteElement).toBeVisible()
        expect(quoteElement).toContainElement(quotePhotos[index])
      }
    })
  })

  describe('style tests:', () => {
    it('should render with correct styles', () => {
      const testimonialsTree = create(<Testimonials />).toJSON()
      expect(testimonialsTree).toMatchSnapshot()
    })
  })
})
