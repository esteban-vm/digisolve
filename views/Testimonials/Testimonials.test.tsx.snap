// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª TESTIMONIALS: style tests: should render with correct styles 1`] = `
.emotion-0 {
  background-image: linear-gradient(var(--color-overlay3), var(--color-overlay4)),url('/img/testimonial-bg.jpg');
  -webkit-background-size: cover;
  background-size: cover;
  background-attachment: fixed;
  text-align: center;
  color: var(--color-white);
}

@media only screen and (max-width: 480px) {
  .emotion-0 {
    padding: 0 10%;
  }
}

.emotion-0 h2 {
  font-size: 250%;
  margin-bottom: 20px;
  margin-top: 150px;
  text-transform: capitalize;
}

@media only screen and (max-width: 1200px) {
  .emotion-0 h2 {
    margin-top: 90px;
  }
}

@media only screen and (max-width: 768px) {
  .emotion-0 h2 {
    margin-top: 60px;
    margin-bottom: 10px;
  }
}

.emotion-1 {
  text-align: left;
  margin-bottom: 50px;
}

@media only screen and (max-width: 480px) {
  .emotion-1 {
    margin-bottom: 5px;
  }
}

.emotion-1 blockquote {
  padding: 2%;
  font-style: italic;
  line-height: 145%;
  position: relative;
  margin-top: 40px;
}

.emotion-1 blockquote::before {
  content: '\\201C';
  display: block;
  font-size: 500%;
  position: absolute;
  left: -5px;
  top: -5px;
}

.emotion-1 blockquote::first-letter {
  text-transform: capitalize;
}

.emotion-1 cite {
  display: block;
  margin-top: 25px;
  font-size: 90%;
  text-transform: capitalize;
}

.emotion-1 img {
  border-radius: 50%;
  height: 20%;
  vertical-align: middle;
  margin-right: 10px;
}

<section
  aria-labelledby=":r0:"
  className="emotion-0"
  id="testimonials"
>
  <div
    className="row "
    role="presentation"
  >
    <h2
      id=":r0:"
    >
      Our testimonials
    </h2>
  </div>
  <div
    className="row "
    role="presentation"
  >
    <div
      className="col span_1_of_3 "
      role="presentation"
    >
      <article
        aria-labelledby="c2f0fc36-aad3-4e98-9ff8-61a30257d41c"
        className="emotion-1"
      >
        <blockquote>
          Voluptates hic pariatur reiciendis minus vel labore dolorum.
        </blockquote>
        <cite>
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={72}
            id="c2f0fc36-aad3-4e98-9ff8-61a30257d41c"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2Fclient-1.jpg&w=256&q=75"
            srcSet="/_next/image?url=%2Fimg%2Fclient-1.jpg&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fclient-1.jpg&w=256&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={72}
          />
          Herman Gibson
        </cite>
      </article>
    </div>
    <div
      className="col span_1_of_3 "
      role="presentation"
    >
      <article
        aria-labelledby="efc0be82-fe34-4df9-8821-cf195a75da1d"
        className="emotion-1"
      >
        <blockquote>
          Voluptates hic pariatur reiciendis minus vel labore dolorum.
        </blockquote>
        <cite>
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={72}
            id="efc0be82-fe34-4df9-8821-cf195a75da1d"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2Fclient-2.jpg&w=256&q=75"
            srcSet="/_next/image?url=%2Fimg%2Fclient-2.jpg&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fclient-2.jpg&w=256&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={72}
          />
          Peter Kemmer
        </cite>
      </article>
    </div>
    <div
      className="col span_1_of_3 "
      role="presentation"
    >
      <article
        aria-labelledby="c679bf1c-1387-4aa0-9eec-ab2d28d79310"
        className="emotion-1"
      >
        <blockquote>
          Voluptates hic pariatur reiciendis minus vel labore dolorum.
        </blockquote>
        <cite>
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={72}
            id="c679bf1c-1387-4aa0-9eec-ab2d28d79310"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2Fclient-3.jpg&w=256&q=75"
            srcSet="/_next/image?url=%2Fimg%2Fclient-3.jpg&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fclient-3.jpg&w=256&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={72}
          />
          Naomi Stoltenberg
        </cite>
      </article>
    </div>
  </div>
</section>
`;
